Data/Base/Shaders/Common/ObjectConstants.h

```diff
@@ -23,7 +23,7 @@ StructuredBuffer<ezPerInstanceData> perInstanceData;
 StructuredBuffer<Transform> skinningTransforms;
 #  endif
 
-Buffer<float4> perInstanceVertexColors;
```

Comment:
This is not the same, the underlying buffer is one uint per element and we utilize the hardware to do the conversion to float4 for us. StructuredBuffer<float4> indicates that the underlying buffer already is a float4 per element, which is not the case.

Reply:
Ah ok, didn't know such a thing existed. Afaik WebGPU doesn't have this feature. So the correct change would be to use a StructuredBuffer<uint32> and then unpack and rescale to float4 in the shader? Do you think that would make a performance difference? I'd assume D3D11 just injects such code into the shader for you, or is that indeed dedicated hardware ?

Reply:
A StructuredBuffer<uint> is not a good idea, the struct in a structured buffer shouldn't be that small as a single uint, 16bytes at least for good perf, ideally also 16bytes aligned. Would Buffer<uint> work on WebGPU or does it not support raw buffers at all? 
No D3D11 does not inject shader code, maybe the driver does, but in most cases it will just utilize the same HW that is used for texture fetches which already have conversions like these build in. In theory we could also use a texture but then you have stricter size limitations, so Buffer<uint> with manual conversion in the shader would be the best solution if it is supported.

Reply:
IIRC the Spir-V to WGSL compiler couldn't handle this. So could be a not implemented feature, or something that has no equivalent there. WebGPU has storage buffers, which are, to my knowledge just raw buffers. I think they should be used, but not sure what change in HLSL would be necessary.

I'll probably revert some of these changes for now, and come back to them later, but now I at least already know better what to do about them.

Reply:
Storage buffers are structured buffers, here is a discussion regarding storage **texel** buffers (that's the Buffer equivalent in vulkan or opengl) in WebGPU: https://github.com/gpuweb/gpuweb/issues/162
TLDR they are not supported yet.

Reply:
Removed from this PR.

Reply:
Added `ezGALDeviceCapabilities::m_bTexelBuffer` and platform define `SUPPORTS_TEXEL_BUFFER`. These must match and are static per platform as the later one is defined in the shader platform headers. Implementation passes proc gen test if forced on.